core/vm/context.go

```diff
@@ -75,13 +75,13 @@ type ChainContext interface {
 }
 
 // NewEVMContext creates a new context for use in the EVM.
-func NewEVMContext(msg Message, header *types.Header, chain ChainContext, author *common.Address) Context {
-	// If we don't have an explicit author (i.e. not mining), extract from the header
+func NewEVMContext(msg Message, header *types.Header, chain ChainContext, txFeeRecipient *common.Address) Context {
+	// If we don't have an explicit txFeeRecipient (i.e. not mining), extract from the header
 	var beneficiary common.Address
-	if author == nil {
-		beneficiary, _ = chain.Engine().Author(header) // Ignore error, we're past header validation
+	if txFeeRecipient == nil {
```

Comment:
this looks fishy... if the header has a coinbase (which might be set to validator address in PREFORK status) here we are not using that information and just using the configured txFeeRecipient. 

I would investigate this

Reply:
I don't get why if the header has a coinbase we ignore it. @kevjue ?

Reply:
Ya, agreed that this looks fishy.  I haven't been following this issue.  What's the semantics of the "coinbase" field in the header after this change?  @mcortesi @andres-dg

Reply:
coinbase is the address where gas fees are deposited; so that's been called "txFeeRecipient" and then we have another address for the header signer (the validator address)

Now, since this is a hardfork, we need to make sure that before activation the "coinbase is always the validator" and only in the case of hardfork activation they can be different

Reply:
OK.  To answer your original question, I suspect that txFeeRecipient is used ONLY when this function is called by the proposer when creating a block proposal, and for all other times txFeeRecipient is nil.  So that would explain why the coinbase field is ignored.  If that's true, then the code here is obviously not clear about that and seems like at the very least, a comment should be added.